framework/src/engine/bft/method.ts

```diff
@@ -217,7 +217,11 @@ export class BFTMethod {
 			throw new Error('Invalid certificateThreshold input.');
 		}
 
-		const validatorsHash = computeValidatorsHash(validators, certificateThreshold);
+		sortValidatorsByAddress(validators);
+		const validatorsHash = computeValidatorsHash(
```

Comment:
I think there is still a discrepancy between LIP and implementation: The LIP assumes that the input of `computeValidatorsHash`  is an array with items of type (bftWeight, blsKey). In the implementation however,  you call computeValidatorsHash with corresponding argument validators (after filtering bft weight 0) , which is an array with items of type (address, bftWeight, generatorKey, blsKey) . 

Not a big deal, it would be good to change it in order to be fully consistent, but since it seems to work correctly now, it's ok to leave it as is..

Reply:
Updated `computeValidatorsHash ` aligning with LIP